.rubocop.yml

```diff
@@ -0,0 +1,129 @@
+require: rubocop-rspec
+
+inherit_mode:
+  merge:
+    - Include
+    - Exclude
+
+AllCops:
+  Include:
+    - '**/Rakefile'
+    - '**/config.ru'
+    - 'lib/'
+  Exclude:
+    - 'db/**/*'
+    - 'bin/bundle'
+
+Rails:
+  Enabled: true
```

Comment:
We don't want this for every repo.

Reply:
> Can you be more specific about what was removed from the Format config?

```diff
*** .rubocop.yml	2019-04-24 12:09:47.000000000 -0400
--- ../styleguide/.rubocop.yml	2019-04-25 17:36:15.000000000 -0400
***************
*** 3,20 ****
  inherit_mode:
    merge:
      - Include
  
  AllCops:
-   TargetRailsVersion: 5.2
-   TargetRubyVersion: 2.3
    Include:
      - '**/Rakefile'
      - '**/config.ru'
      - 'lib/'
    Exclude:
      - 'db/**/*'
-     - 'gem-themes-core/'
-     - 'gem-themebuilder/'
      - 'bin/bundle'
  
  Rails:
--- 3,17 ----
  inherit_mode:
    merge:
      - Include
+     - Exclude
  
  AllCops:
    Include:
      - '**/Rakefile'
      - '**/config.ru'
      - 'lib/'
    Exclude:
      - 'db/**/*'
      - 'bin/bundle'
  
  Rails:
***************
*** 29,35 ****
      - 'context'
      - 'concern'
      - 'concerning'
-     - 'FactoryBot.define'
      - 'class_methods'
      - 'included'
      - 'state_machine'
--- 26,31 ----
***************
*** 69,84 ****
    Enabled: false
  
  Rails/UnknownEnv:
!   Environments: [ production, staging, development, experimental, test ]
  
  Style/EmptyMethod:
    EnforcedStyle: expanded
  
- Style/GlobalVars:
-   AllowedVariables:
-     - $flipper
-     - $REDIS
- 
  Layout/MultilineMethodCallIndentation:
    EnforcedStyle: indented
    IndentationWidth: 2
--- 65,75 ----
    Enabled: false
  
  Rails/UnknownEnv:
!   Environments: [ production, staging, development, test ]
  
  Style/EmptyMethod:
    EnforcedStyle: expanded
  
  Layout/MultilineMethodCallIndentation:
    EnforcedStyle: indented
    IndentationWidth: 2
***************
*** 136,142 ****
  
  RSpec/ExampleLength:
    Max: 10
- 
- # TODO Change from Syck to Psych for YAML parsing so that we can use this.
- Security/YAMLLoad:
-   Enabled: false
--- 127,129 ----

```

> How do you plan to use this config in individual repos? From what I remember, RuboCop has an import system, but so does Code Climate.

The plan is to use the `inherit_from` with a raw GitHub http link as documented here: https://rubocop.readthedocs.io/en/latest/configuration/#inheriting-configuration-from-a-remote-url

This should also work for CodeClimate, otherwise we can modify the CodeClimate prepare step: https://docs.codeclimate.com/docs/configuring-the-prepare-step

Reply:
> Can you be more specific about what was removed from the Format config?

```diff
*** .rubocop.yml	2019-04-24 12:09:47.000000000 -0400
--- ../styleguide/.rubocop.yml	2019-04-25 17:36:15.000000000 -0400
***************
*** 3,20 ****
  inherit_mode:
    merge:
      - Include
  
  AllCops:
-   TargetRailsVersion: 5.2
-   TargetRubyVersion: 2.3
    Include:
      - '**/Rakefile'
      - '**/config.ru'
      - 'lib/'
    Exclude:
      - 'db/**/*'
-     - 'gem-themes-core/'
-     - 'gem-themebuilder/'
      - 'bin/bundle'
  
  Rails:
--- 3,17 ----
  inherit_mode:
    merge:
      - Include
+     - Exclude
  
  AllCops:
    Include:
      - '**/Rakefile'
      - '**/config.ru'
      - 'lib/'
    Exclude:
      - 'db/**/*'
      - 'bin/bundle'
  
  Rails:
***************
*** 29,35 ****
      - 'context'
      - 'concern'
      - 'concerning'
-     - 'FactoryBot.define'
      - 'class_methods'
      - 'included'
      - 'state_machine'
--- 26,31 ----
***************
*** 69,84 ****
    Enabled: false
  
  Rails/UnknownEnv:
!   Environments: [ production, staging, development, experimental, test ]
  
  Style/EmptyMethod:
    EnforcedStyle: expanded
  
- Style/GlobalVars:
-   AllowedVariables:
-     - $flipper
-     - $REDIS
- 
  Layout/MultilineMethodCallIndentation:
    EnforcedStyle: indented
    IndentationWidth: 2
--- 65,75 ----
    Enabled: false
  
  Rails/UnknownEnv:
!   Environments: [ production, staging, development, test ]
  
  Style/EmptyMethod:
    EnforcedStyle: expanded
  
  Layout/MultilineMethodCallIndentation:
    EnforcedStyle: indented
    IndentationWidth: 2
***************
*** 136,142 ****
  
  RSpec/ExampleLength:
    Max: 10
- 
- # TODO Change from Syck to Psych for YAML parsing so that we can use this.
- Security/YAMLLoad:
-   Enabled: false
--- 127,129 ----

```

> How do you plan to use this config in individual repos? From what I remember, RuboCop has an import system, but so does Code Climate.

The plan is to use the `inherit_from` with a raw GitHub http link as documented here: https://rubocop.readthedocs.io/en/latest/configuration/#inheriting-configuration-from-a-remote-url

This should also work for CodeClimate, otherwise we can modify the CodeClimate prepare step: https://docs.codeclimate.com/docs/configuring-the-prepare-step

Reply:
Removed.